zerver/lib/test_runner.py

```diff
@@ -491,9 +500,36 @@ def run_tests(self, test_labels: List[str],
         result = self.run_suite(suite)
         self.teardown_test_environment()
         failed = self.suite_result(suite, result)
-        if not failed:
+        failed_thread = []  # type: List[str]
+        if not failed or self.parallel == 1:
             write_instrumentation_reports(full_suite=full_suite, include_webhooks=include_webhooks)
-        return failed, result.failed_tests
+            return failed, result.failed_tests, failed_thread
+
+        # Below is the logic for passing the list of tests ran in a failed
+        # worker, for use in the --replicate option when running test-backend.
+        args = [
+            (index, subsuite[1])
+            for index, subsuite in enumerate(suite.subsuites)
+        ]  # type: List[Tuple[int, List[str]]]
+
+        # We iterate through the subsuites for the failure. When
+        # we find it, assign the index for use in aggregating the
+        # individual tests.
+        index = None
```

Comment:
I think this index variable is unnecessary; we should just use `failed_pid = None` as initialization and save some complexity.